src/plugins/discover/public/application/angular/context/api/anchor.js

```diff
@@ -41,7 +41,10 @@ export function fetchAnchorProvider(indexPatterns, searchSource) {
         language: 'lucene',
       })
       .setField('sort', sort);
-
+    if (useNewFieldsApi) {
+      searchSource.removeField('fieldsFromSource');
```

Comment:
Could you shortly explain me, in what case we would come here with `useNewFieldsApi == true` and `fieldsFromSource` has been set? Looking at the rest of the code, it looks like when we're using this flag, we will always already set the fields instead of the fieldsFromSource?

Reply:
If you enable `useNewFieldsApi`, then disable it, then enable it again - `fieldsFromSource` will still be set on search source and the request will include `_source` fields.